types/sqs-consumer/index.d.ts

```diff
@@ -1,29 +1,34 @@
-// Type definitions for sqs-consumer 3.5
+// Type definitions for sqs-consumer 3.7
 // Project: https://github.com/BBC/sqs-consumer
-// Definitions by: Daniel Chao <http://dchao.co/>, Eric Byers <http://www.github.com/EricByers>
+// Definitions by: Daniel Chao <http://dchao.co/>
+//                 Eric Byers <http://github.com/EricByers>
+//                 Ezinwa Okpoechi <http://github.com/BrainMaestro>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 import { SQS } from "aws-sdk";
 
-export type ConsumerDone = (error?: Error) => void;
+export type ConsumerDone = (error?: SQSError) => void;
 
 export interface Options {
-  queueUrl: string;
-  handleMessage(message: SQS.Message, done: ConsumerDone): any;
-  region?: string;
-  attributeNames?: string[];
-  messageAttributeNames?: string[];
-  batchSize?: number;
-  visibilityTimeout?: number;
-  waitTimeSeconds?: number;
-  authenticationErrorTimeout?: number;
-  sqs?: SQS;
+    queueUrl: string;
+    handleMessage(message: SQS.Message, done: ConsumerDone): any;
+    region?: string;
+    attributeNames?: string[];
+    messageAttributeNames?: string[];
+    batchSize?: number;
+    visibilityTimeout?: number;
+    waitTimeSeconds?: number;
+    authenticationErrorTimeout?: number;
+    sqs?: SQS;
 }
 
-export interface Consumer extends NodeJS.EventEmitter {
-  start(): void;
-  stop(): void;
+declare class SQSError extends Error {}
```

Comment:
I would leave out this class declaration if it doesn't provide anything special.

TypeScript type-checks shapes of objects, and to TS, `SQSError` and `Error` are identical.

Reply:
I agree that they are identical to TS, but you can still differentiate between them with `instanceof`. The library throws an `SQSError` in some cases, and having this declaration together with the `instanceof` check can allow you to handle differently from other errors.

Also, just to represent the library as completely as possible

Reply:
Ah, that's a good point. This sounds fair then.

Reply:
Actually, you can't use `instanceof` here because the constructor is never exported. But nevertheless, I don't actually care too much whether we're declaring an `SQSError` class or not anyway. I'm good with this being here.

Reply:
`declare` was actually exporting the class. Might be a little non-standard. I changed to `export` though